core/eolearn/core/utils/parsing.py

```diff
@@ -13,15 +13,39 @@
 """
 from __future__ import annotations
 
+import sys
 from itertools import repeat
-from typing import TYPE_CHECKING, Iterable, List, Optional, Sequence, Tuple, Union, cast
+from typing import TYPE_CHECKING, Dict, Iterable, List, Optional, Sequence, Tuple, Union, cast
 
 from ..constants import FeatureType
+from .types import EllipsisType
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal  # pylint: disable=ungrouped-imports
 
 if TYPE_CHECKING:
     from ..eodata import EOPatch
 
 
+FeatureSpec = Union[Tuple[Literal[FeatureType.BBOX, FeatureType.TIMESTAMP], None], Tuple[FeatureType, str]]
```

Comment:
Would `[FeatureType.TIMESTAMP, 'something']` pass? Is that an issue if it does?

Reply:
It would. I don't see much issue with that, otherwise the types get incredibly specific to the point of unreadable :(
One of the core issues with feature parsing is the fact that timestamps and bbox need to be unique. Probably a design flaw that they are treated as features at all.

Reply:
Yes, I agree, these exceptions are causing headaches. perhaps some similar approach to datacubes should be considered, where timestamps and bbox are treated as coordinates (t, x, y). 

Resolving.